### DISCUSSION TITLE
Cant get acces to puncover webserver running in Docker container

### DISCUSSION THREAD

**Discussion by ArkadiuszGrzelka** [zephyrproject-rtos/zephyr#68270] [General] [answered]
Hello

I have quite intresting problem. 
To development, I'm using docker image, all works fine except puncover.
I Added puncover to docker image by:

```
RUN pip install puncover
EXPOSE 5000
```

run docker with port mirroring: 
`-p 5001:5000/tcp`

And after run puncover, 
`west build -t puncover -d "${PROJECT_BUILD_DIR}"`

I got:
```
* Serving Flask app 'puncover.puncover' (lazy loading)
* Environment: production
WARNING: This is a development server. Do not use it in a production deployment.
Use a production WSGI server instead.
* Debug mode: off
* Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
```

Everything seems fine, but I can't get access to puncover output:
 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/143822370/a37d29b3-602e-4253-941f-1ba33a041d6e)

Do you have any ideas what is wrong?
My operating system is Ubuntu 22.04

**Comment by kartben** [comment, marked as answer]
I think the server would need to listen on 0.0.0.0 to be visible outside of the container. See `--host 0.0.0.0` CLI option.
Hope this helps :)

**Reply by ArkadiuszGrzelka** [reply]
@kartben thanks for answer.
I solved my problem ;) 
Thanks a lot.

**Reply by kartben** [reply]
Great!
